components/scream/src/physics/p3/atmosphere_microphysics.hpp

```diff
@@ -53,6 +68,85 @@ class P3Microphysics : public AtmosphereProcess
   const std::set<FieldIdentifier>& get_required_fields () const { return m_required_fields; }
   const std::set<FieldIdentifier>& get_computed_fields () const { return m_computed_fields; }
 
+  /*--------------------------------------------------------------------------------------------*/
+  // Most individual processes have a pre-processing step that constructs needed variables from
+  // the set of fields stored in the field manager.  A structure like this defines those operations,
+  // which can then be called during run_imple in the main .cpp code.
+  // Structure to handle the local generation of data needed by p3_main in run_impl
+  struct run_local_vars {
+    run_local_vars(const int ncol, const int npack, view_2d_const pmid_, view_2d T_atm_, view_2d_const ast_, view_2d_const zi_) : 
+      m_ncol(ncol),
+      m_npack(npack),
+      // IN
+      pmid(pmid_),
+      T_atm(T_atm_),
+      ast(ast_),
+      zi(zi_),
+      // OUT
+      exner("exner",ncol,npack),
+      th_atm("th_atm",ncol,npack),
+      cld_frac_l("cld_frac_l",ncol,npack),
+      cld_frac_i("cld_frac_i",ncol,npack),
+      cld_frac_r("cld_frac_r",ncol,npack),
+      dz("dz",ncol,npack)
+    {
+      // Nothing else to initialize at the moment.
+    };
+    KOKKOS_INLINE_FUNCTION
+    void operator()(const int icol) const {
+      int ipack, ivec;
+      for (ipack=0;ipack<m_npack;ipack++) {
+        // Exner
+        const Spack opmid(pmid(icol,ipack));
+        const Smask opmid_mask(!isnan(opmid) and opmid>0.0);
+        auto oexner = physics_fun::get_exner(opmid,opmid_mask);
```

Comment:
I'm glad you enjoy so much the physics that you call it 'fun'. :)

Reply:
yeah man, got make people excited to work with the `p3` code.  In a more serious response, it turns out that `physics` already has a defined meaning in `atmosphere_microphysics.hpp` so I had to add the fun.